src/identity/configuration/IdentityProviderFactory.ts

```diff
@@ -389,13 +391,17 @@ export class IdentityProviderFactory implements ProviderFactory {
 
       this.logger.warn(`OIDC request failed: ${detailedError}`);
 
+      let resultingError: HttpError = new OAuthHttpError(out, error.name, oidcError.statusCode, error.message);
+      // Keep the original stack to make debugging easier
+      resultingError.stack = error.stack;
+
       // OIDC library hides extra details in these fields
       if (this.showStackTrace) {
-        error.message = detailedError;
+        resultingError.message = detailedError;
 
         // Also change the error message in the stack trace
-        if (error.stack) {
-          error.stack = error.stack.replace(/.*/u, `${error.name}: ${error.message}`);
+        if (resultingError.stack) {
+          resultingError.stack = resultingError.stack.replace(/.*/u, `${error.name}: ${error.message}`);
```

Comment:
I think all of the above lines should be done by the `OAuthHttpError` constructor though, which just takes `cause` as an argument.

Reply:
No need to make it more complex to implement my suggestion.